docs/udf.rst

```diff
@@ -317,7 +317,58 @@ To invoke a UDF like this, the apply_neighborhood method is most suitable:
             {'dimension': 'y', 'value': 128, 'unit': 'px'}
         ], overlap=[])
 
+Inspecting variables within UDF
+========================================
+
+To print and inspect variables that are within the UDF, users can use `inspect(data=[], message="")` function. 
+This will print the data that is supplied within, and show it with the message within the logs.
+
+.. code-block:: python
+    :linenos:
+    :caption: ``Inspecting UDFs``
+    :emphasize-lines: 7
+
+    # Create a UDF object from inline source code.
+    udf = openeo.UDF("""
+    import xarray
+
+    def apply_datacube(cube: xarray.DataArray, context: dict) -> xarray.DataArray:
+        cube.values = 0.0001 * cube.values  
+        inspect(data=[type(cube.values)], message="The dtype of cube.values")
+        return cube
+    """)
+
+In the above example, the `inspect` function is used to retrieve the datatype of `cube.values`. Once the job logs are opened in the Web Editor, the result will appear
+under the supplied message. This case it will be shown that `Data: <class 'numpy.ndarray'>`
+ 
+Passing user defined variables to UDF
+========================================
+
+In order to pass variables and values that are used throughout the user side of script, these need to be put in the `context` dictionary.
```

Comment:
> used throughout the user side of script

I'm not sure what you mean here, e.g. with "user side". And what "script" are you referring to? The script that build the process graph, or the UDF script?

Reply:
I'm still a bit confused what you mean:
> where the user runs the UDF.

the user does not run the UDF user side, it's the backend that executes the UDF backend-side

Reply:
Sorry, the script where the user **defines** the UDF script.